src/controllers/user.controller.js

```diff
@@ -0,0 +1,90 @@
+const userService = require('./../services/user.service');
+
+const getUsers = async (req, res) => {
+  try {
+    const users = await userService.getAllUsers();
+
+    res.send(users);
+  } catch (error) {
+    res.status(500).send('Server error');
+  }
+};
+
+const getOneUser = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const user = await userService.getUserById(id);
+
+    if (!user) {
+      return res.status(404).send('User not found');
+    }
+    res.send(user);
+  } catch (error) {
+    res.status(500).send('Server error');
+  }
+};
+
+const createUser = async (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    return res.status(400).send('Name is required');
+  }
+
+  try {
+    const user = await userService.createUser(name);
+
+    res.status(201).send(user);
+  } catch (error) {
+    res.status(500).send('Server error');
+  }
+};
+
+const updateUser = async (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+
+  try {
+    const user = await userService.getUserById(id);
+
+    if (!user) {
+      return res.status(404).send('User not found');
+    }
+
+    if (typeof name !== 'string' || name.length === 0) {
+      return res.status(422).send('Invalid name');
+    }
+
+    const updatedUser = await userService.updateUser({ id, name });
+
+    res.status(200).send(updatedUser);
+  } catch (error) {
+    res.status(500).send('Server error');
+  }
+};
+
+const removeUser = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const user = await userService.getUserById(id);
+
+    if (!user) {
+      return res.status(404).send('User not found');
+    }
+
+    await userService.removeUser(id);
+    res.status(204).send();
+  } catch (error) {
+    res.status(500).send('Server error');
```

Comment:
Consider adding more detailed error messages or logging the error details in the catch block to help with debugging and monitoring.